mathics/builtin/system.py

```diff
@@ -345,10 +345,9 @@ def evaluate(self, evaluation):
         except ValueError:
             # not run in script mode
             return Expression(SymbolList)
-
-        return Expression(
-            SymbolList, *(String(arg) for arg in sys.argv[dash_index + 1 :])
-        )
+        scriptname = "" if dash_index == 0 else sys.argv[dash_index - 1]
```

Comment:
This seems totally independent. What's up here? 

This probably should go into a separate PR.

Reply:
@rocky,
the issue contained several aspects. 
- How `FindRoot` handles expressions which can not have an analytic derivative
- How to evaluate derivatives of `NIntegral`
- How we are processing command line parameters.

So I put here all together to see if it solves the test case proposed.

In this particular module, I proposed a fixed for the last one.

Reply:
Ok. But let's me clear. Users and issues will sometimes be something like: please fix everything that I just encountered.

We should try to split up the issues so that each covers one thing so that thing can be tracked. And similarly in PRs it would be nice to split separate problems into separate PRs for the same reason. They will get through faster. 

And in 6 months when we are looking over looking at a particular line of code and are wondering why it is there and then look at the commit or merge associated with it, it reflects one idea, and not several.